source/client/cl_steam.c

```diff
@@ -104,5 +104,12 @@ void Steam_RequestAvatar(uint64_t steamid, int size)
 */
 void Steam_AdvertiseGame( const uint8_t *ip, unsigned short port )
 {
-	// UNIMPLEMENTED_DBGBREAK();
+	char* keys[1] = {"connect"};
+	char* values[1];
+	if (port) {
+		values[0] = va("+connect %d.%d.%d.%d:%i",ip[0],ip[1],ip[2],ip[3],port);
```

Comment:
this is on the main thread right? va is not a thread safe operation.

Reply:
how so?

Reply:
Because the function is called by all threads, and it retains state.

Reply:
btw not a lot in this engine is.
There is also stuff like `NET_AddressToString`, but that is also not threadsafe.